src/main/java/Duke.java

```diff
@@ -1,10 +1,148 @@
+import java.util.Scanner;
 public class Duke {
+    private static final int MAX_TASK_LENGTH = 100;
+    private static int taskCount = 0;
+
     public static void main(String[] args) {
+        //Print logo and opening message
+        printLogo();
+        printGreeting();
+
+        //Receive user input and execute corresponding functions
+        getUserInput();
+
+        //Print closing message
+        printFarewell();
+    }
+
+    public static void loopEcho() {
+        Scanner sc = new Scanner(System.in);
+        String userInput = sc.nextLine();
+        while (!userInput.equals("bye")) {
+            printDivider();
+            System.out.printf("\t%s%n", userInput);
+            printDivider();
+            userInput = sc.nextLine();
+        }
+    }
+
+    public static void getUserInput() {
+        Task[] taskList = new Task[MAX_TASK_LENGTH];
+        int listIndex = 0;
+        Scanner sc = new Scanner(System.in);
+        String userInput = sc.nextLine();
+        while (!userInput.trim().toLowerCase().equals("bye")) {
+            if (userInput.trim().toLowerCase().equals("list")) {
+                displayList(taskList);
+            } else if (userInput.trim().toLowerCase().contains("done")) {
+                markTaskDone(taskList, userInput);
+            } else if (userInput.trim().toLowerCase().contains("todo")) {
+                listIndex = addToList(taskList, listIndex, userInput, "todo");
+            } else if (userInput.trim().toLowerCase().contains("deadline")) {
+                listIndex = addToList(taskList, listIndex, userInput, "deadline");
+            } else if (userInput.trim().toLowerCase().contains("event")) {
+                listIndex = addToList(taskList, listIndex, userInput, "event");
+            } else {
```

Comment:
To reduce the number of branches both in the while loop and the addToList method, you can consider just modifying the addToList method to simply taking in the taskList and a Task. Above each addToList, separate methods to create each task object eg. deadline/event can be created to handle the user input. This way, the addToList method will not be handling too many different functions on its own, and we can make use of polymorphism to simplify the method.